### PR TITLE
feat(repo): add the ATCvet medication spine and show cross-vocabulary codes

### LINKED ISSUE #2593: PIMS: show VeNom/SNOMED equivalents in the SOAP coded-term picker
## Summary

Show the VeNom/SNOMED equivalents of a clinical term in the SOAP coded-term picker, and record them with the pick.

## Problem / motivation

#2496 added the picker and #2501 made picked terms export as VeNom/SNOMED CodeableConcepts, but that projection lives entirely in the FHIR/API layer. In the UI a clinician still sees only `YC-005784`, so there is no way to tell — while choosing — whether a term carries a SNOMED equivalent, or whether that equivalent is exact. The vocabulary work is invisible at the point of use.

## Proposed solution

- `GET /v1/codes/terms/suggest` returns a `codings[]` per suggestion: the strongest usable crosswalk per system, read from `CodeMapping` with the same usable-equivalence gate the FHIR export applies, in one batched query per result page.
- The picker dropdown states each crosswalk beside the YC code, and chips carry them as badges.
- A non-exact equivalence is labelled (`SNOMED 422400008 (narrower)`) so a broader/narrower match is never read as the same concept.
- Picked crosswalks are stored on the term so a note shows the codes the clinician actually saw; the export still projects live from `CodeMapping`.

## Acceptance criteria

- Dropdown and chips show VeNom/SNOMED codes where they exist, nothing where they do not.
- What the picker advertises is exactly what the export would emit (same gate, same ranking).
- Inexact equivalences are visibly qualified.
- One crosswalk query per keystroke-page, not per suggestion.
- Malformed stored codings are dropped on read; unmapped terms carry no `codings` key.

## Test plan

Backend: batched query shape, gate, strongest-per-system ranking, empty page. Frontend: dropdown rendering, chip badges, carry-onto-pick, unknown-system fallback, parser validation/cap. All mutation-checked.

### LINKED ISSUE #2600: PIMS: ATCvet medication spine — import, search, formulary links and coded prescribing
## Summary

Add the ATCvet drug classification as Yosemite Crew's medication spine: import the WHO CC index, search it, link a practice's formulary and stock to it, and let clinicians prescribe a coded substance.

## Problem / motivation

Medications are free text everywhere. `DrugFormulary.drugName`, `InventoryItem.name` and prescription lines are strings, so nothing can answer "which patients received a systemic antibacterial", nothing exports a coded prescription, and no interaction or stewardship work has a foundation. The clinical vocabulary programme coded diagnoses and procedures; medications were the remaining gap.

## Proposed solution

- Import all 8,315 ATCvet codes as a new `ATCVET` code system (6,417 substances as `MEDICATION`, 1,898 grouping levels as `MEDICATION_CATEGORY`), with the classification's five levels as `CodeRelationship` edges.
- `GET /v1/codes/medications/suggest` (web + mobile): substance search with each result carrying its class path, species (immunologicals only) and an antibacterial flag.
- `atcCode` on `DrugFormulary` and `InventoryItem`, plus a conservative backfill.
- ATCvet substances offered in the prescription search box, and prescriptions exported with a real `medicationCodeableConcept.coding`.

## Scope

Backend import/search, schema links, backfill, and the prescribing picker.

### Out of scope

Dose/DDD data (a separate WHO CC product), interaction checking, stewardship reporting UI, and mapping ATCvet to other drug vocabularies.

## Acceptance criteria

- Import is idempotent, dry-run by default, and reports every skipped row with a reason.
- Only substances are prescribable; grouping levels never appear in search.
- Backfill never guesses: ambiguous or unmatched drugs stay uncoded and are reported.
- A coded prescription exports an ATCvet coding; an uncoded one carries text only.
- Medication search is index-backed, not a sequential scan.

## Test plan

Unit tests over the importer's planner, the search query builder and service, the backfill matcher, the controller, and the prescribing picker; all mutation-checked. Live verification on dev after deploy.

## Dependencies & risks

The ATCvet index is licensed from the WHO CC; License.txt gains a third-party data notice. Yearly releases are re-imported with the same idempotent script.

### PR DESCRIPTION
**Pull request by ankit-yc**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/YosemiteCrew/Yosemite-Crew/blob/main/CONTRIBUTING.md#commit-message-convention.
- [x] There is an issue for the bug/feature this PR is for.
- [x] All existing tests and lints pass.

## What is the current behavior?

Two gaps, one reported and one structural.

**Reported:** the SOAP coded-term picker showed Yosemite codes only. #2501 made picked terms *export* as VeNom/SNOMED CodeableConcepts, but that projection lived in the FHIR layer, so a clinician choosing a term could not see whether it had a SNOMED equivalent or how exact it was.

**Structural:** medications were free text everywhere — `DrugFormulary.drugName`, `InventoryItem.name`, prescription lines. Nothing could answer "which patients received a systemic antibacterial", and no prescription left the system coded.

### Cross-codes in the picker (closes #2593)

`terms/suggest` now returns each term's crosswalks — strongest per system, read from `CodeMapping` under the *same* usable-equivalence gate the FHIR export applies, in one batched query per result page, so the picker can never advertise a code the export would refuse to emit. Dropdown rows and chips show them, and a non-exact equivalence is stated rather than hidden: `SNOMED 422400008 (narrower)`. Picked crosswalks are stored with the term so a note shows what the clinician actually saw; the export still projects live.

### ATCvet medication spine (closes #2600)

- **Import**: all 8,315 codes as the new `ATCVET` system — 6,417 substances (`MEDICATION`) and 1,898 grouping levels (`MEDICATION_CATEGORY`) — with 8,300 `is a` edges. The hierarchy needs no parent table: an ATCvet code contains its ancestry in its own prefixes. Idempotent, dry-run by default, every skip reported. A converter regenerates the JSON from each yearly workbook.
- **Search**: `GET /v1/codes/medications/suggest` (web + mobile). Substances only; each result carries its class path (`doxycycline · Tetracyclines · ANTIBACTERIALS FOR SYSTEMIC USE`) resolved in one batched lookup, plus species and an `antibacterial` flag for QJ01.
- **Links**: `atcCode` on `DrugFormulary` and `InventoryItem`, with a backfill that matches on the generic name first and **never guesses** (below).
- **Prescribing**: ATCvet substances appear in the prescription search box beneath the practice's own catalogue, and a coded prescription exports a real `medicationCodeableConcept.coding` under `http://www.whocc.no/atcvet`.

## Domain decisions the data forced

Three findings changed the design, each pinned by a test that fails if it is undone:

1. **ATCvet uses INN spellings.** A practice's "Cephalexin" matches nothing — the substance is `cefalexin`. Rather than fuzzy-matching drug names (where a near miss is a *different medicine*), unmatched rows are reported for a human.
2. **A substance holds several codes.** Ibuprofen is `QC01EB16`/`QG02CC01`/`QM01AE01`; doxycycline is oral-dental *and* systemic. "First match wins" would file a drug under a therapeutic class it does not belong to — exactly what stewardship reporting would then read. Ambiguous rows stay uncoded and are named.
3. **QI encodes species, not therapy.** `QI07` is "immunologicals for canidae", so vaccine codes carry species the other fourteen groups do not. Mapped onto our species buckets where the family maps cleanly; `QI20` ("other species") is deliberately left unmapped rather than guessed.

## Two migration traps avoided

- The existing trigram index is **partial to `YOSEMITECODE`**, so ATCvet rows were not covered and medication search would have sequentially scanned every code. Added a matching partial index for `ATCVET`/`MEDICATION`.
- That index had to go in a **separate migration**: PostgreSQL refuses to use a new enum value inside the transaction that added it, and Prisma runs each migration file in one transaction. Combined, it fails on deploy.

## Impact area

Backend terminology services, schema (two nullable columns, three migrations), the SOAP picker and the prescription editor. No changes to existing prescribing behaviour: the ATCvet group sits below the inventory results, and stock prescribing is untouched.

## Validation performed

- Frontend and backend `tsc` clean; scoped ESLint clean; `@yosemite-crew/types` builds.
- Backend: 457 suites / 11,003 tests green. Frontend: the five affected suites, 72 tests green.
- Importer dry run against the real 2026 index: 8,315 codes, 8,300 edges, **zero skips**.
- **Mutation-checked — 24 mutations, each makes exactly its pinning test fail**: equivalence gate and ranking in suggest; inexact-qualifier display; coded-term parsing; importer level typing, dangling-parent guard, code grammar, QI species mapping and dedup; search type filter, LIKE escaping, species-agnostic inclusion, batched ancestors, blank-crumb guard; backfill ambiguity refusal, generic-name preference, leading-number preservation, concurrent-write guard, medicine-only scope; prescribing fulfillment, stale-response guard, minimum query, coding emission. One survivor was investigated and proven an *equivalent* mutant (a provably dead guard), and the dead code removed rather than a test written to fake coverage.
- Secrets scan clean.
- Post-merge: dev deploy, import run, and a workflow QA pass on the picker and prescribing flow with screenshots.

## Related Issue(s)

Closes #2593
Closes #2600

### CODE REVIEW

File: apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx [hunk 1]
```diff
@@ -391,6 +395,47 @@ const PrescriptionRow = ({
...
+        suggestMedications({ q: trimmed, limit: ATCVET_LIMIT })
+          .then((results) => {
+            if (requestSeq.current === requestId) setItems(results);
+          })
+          .catch((error) => {
+            console.error('Unable to suggest medications:', error);
+            if (requestSeq.current === requestId) setItems([]);
+          });
+      },
+      readOnly || query.trim().length < ATCVET_MIN_QUERY ? 0 : ATCVET_DEBOUNCE_MS
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx]: Fixed in e323139: the hook now computes `skip` once from the already-trimmed query and uses it for both the guard and the debounce delay. Also added a test pinning what that ternary is actually for - deleting back below the minimum clears results immediately rather than leaving a stale substance on screen for the debounce window. (My first pass at that test advanced timers past the debounce, so it could not tell the two apart; the mutation survived and I tightened the test rather than claiming it was covered.)

File: apps/backend/src/scripts/import-atcvet-index.ts [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+      continue;
+    }
+    edges.push({ sourceCode: entry.code, targetCode: entry.parent });
+  }
+
+  return { entries, edges, skipped };
+};
+
+const readExtract = (filePath: string): AtcvetExtract =>
+  JSON.parse(fs.readFileSync(path.resolve(filePath), "utf-8"));
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/src/scripts/import-atcvet-index.ts:172]: Adopted in e323139, matching the two sibling importers (import-venom-hierarchy.ts:134 and import-venom-designations.ts:138 carry the identical guard), so this is now consistent with the repo convention rather than a one-off.

Two deliberate differences from the suggested patch. It validated inside readExtract, which sits after an fs.existsSync call - a rejected path would still have been probed for existence first, so the check now runs before any filesystem access. And I updated the doc comment, which previously implied ATCVET_INDEX_PATH could be absolute: paths are repo-relative, matching the guard, so the documentation and the code no longer disagree.

Covered by two tests (traversal and absolute path both rejected), and mutation-checked - removing the guard fails them.

File: apps/frontend/src/app/features/appointments/types/workspace.ts [hunk 3]
```diff
@@ -254,6 +254,13 @@ export type PrescriptionItem = {
   brand?: string;
   /** Generic / composition name from inventory (e.g. "Paracetamol 650"). */
   genericName?: string;
+  /**
+   * ATCvet substance code. Inherited from the inventory item when prescribing
+   * from stock, or set directly when the clinician picks a substance from the
+   * classification. Exported as a FHIR coding so the prescription is readable
+   * outside Yosemite Crew.
+   */
+  atcCode?: string;
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/frontend/src/app/features/appointments/types/workspace.ts:263]: Confirmed and fixed in f749b5e. You were right that it dead-ended: inventoryToPrescriptionItem carried genericName but not the code, so the backfill's work never reached a prescription line - and prescribing from stock is the common path, not the exception. atcCode now flows InventoryItem -> inventoryToPrescriptionItem -> backfillPrescriptionFromInventory, and the DTOs expose it (the list endpoint already spreads the row, so no projection change was needed). Two tests cover it, and removing the mapping fails them.

File: apps/backend/src/scripts/backfill-atc-codes.ts [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  const formularyRows = await prisma.drugFormulary.findMany({
+    where: { atcCode: null },
+    select: { id: true, drugName: true, genericName: true },
+  });
+  const inventoryRows = await prisma.inventoryItem.findMany({
+    // Only medicine stock: coding a lead or a shampoo as a drug is noise.
+    where: {
+      atcCode: null,
+      category: { contains: "medic", mode: "insensitive" },
+    },
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/src/scripts/backfill-atc-codes.ts:144]: Confirmed and fixed. Vaccine is a canonical category in the inventory types, so the contains-medic predicate skipped every vaccine - the QI codes, which are the only ones carrying species, and therefore exactly the rows the species mapping was built for. Now matched against CODEABLE_CATEGORIES (medic, vaccin, drug, pharma) as substrings, so Medicines and Vaccines (canine) are both covered while leads and shampoo stay uncoded. Test asserts both medic and vaccin are in the predicate; reverting to medic-only fails it.

File: apps/frontend/src/app/features/appointments/services/workspaceClinicalService.ts [hunk 5]
```diff
@@ -807,7 +811,22 @@ export const savePrescriptionArtifact = async (
...
+      text: prescription.medicineName,
+      // A coded prescription is readable outside Yosemite Crew; an uncoded one
+      // carries text only rather than a placeholder coding.
+      ...(prescription.atcCode
+        ? {
+            coding: [
+              {
+                system: ATCVET_SYSTEM_URI,
+                code: prescription.atcCode,
+                display: prescription.medicineName,
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/frontend/src/app/features/appointments/services/workspaceClinicalService.ts:824]: Confirmed and fixed - this was the worst of the three, because the client was sending the coding correctly and the server mapper threw it away: medicationCodeableConcept was hardcoded to toCodeableConcept('PRESCRIPTION'). prescriptionToMedicationRequest now rebuilds the concept from the first stored line carrying an atcCode (reading either the line or its metadata), and falls back to the generic concept when nothing is coded rather than inventing a coding. Two mapper tests, mutation-checked.

File: apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx [hunk 6]
```diff
@@ -391,6 +395,47 @@ const PrescriptionRow = ({
   );
 };
 
+const ATCVET_MIN_QUERY = 3;
+const ATCVET_DEBOUNCE_MS = 250;
+const ATCVET_LIMIT = 5;
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx]: Raised 5 -> 10. The cap exists so the classification does not bury the practice's own catalogue above it, but you are right that 5 could truncate a substance's several therapeutic codes - ibuprofen alone holds three. 10 leaves room for those without the ATCvet group dominating the dropdown.

File: apps/backend/src/scripts/import-atcvet-index.ts [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+      batch.map((entry) =>
+        prisma.codeEntry.upsert({
+          where: { system_code: { system: "ATCVET", code: entry.code } },
+          // Re-running a release refreshes the display and provenance rather than
+          // duplicating rows: the import is idempotent by design.
+          update: {
+            display: entry.display,
+            type: entry.type,
+            active: true,
+            meta: meta(entry),
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/src/scripts/import-atcvet-index.ts:233]: Fixed. The importer now deactivates ATCVET entries absent from the extract (active: false) rather than leaving withdrawn codes looking current. Deliberately deactivate and not delete: a prescription or formulary row may already reference the code, and a dangling reference is worse than a retired entry that can still be displayed. The run line now reports the retired count alongside upserts and edges.

File: apps/backend/src/services/atcvet.service.ts [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+    ? Prisma.sql`AND e."meta"->>'atcGroup' = ${group.trim().toUpperCase()}`
+    : Prisma.empty;
+
+  // A substance with no species meta applies to every species; only rows that
+  // name a different species are excluded.
+  const speciesFilter = species
+    ? Prisma.sql`AND (
+        jsonb_typeof(e."meta"->'species') IS DISTINCT FROM 'array'
+        OR e."meta"->'species' @> ${JSON.stringify([species])}::jsonb
+      )`
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/src/services/atcvet.service.ts:97]: Good catch, and subtler than it looks - fixed. 'No species' means two different things in this data: for a general drug it means 'applies to any species' (doxycycline must stay in a feline search), but for an immunological it means 'we refused to guess', which is exactly why QI20 carries none. The filter now keeps species-agnostic rows only when they are not QI, so 'immunologicals for other species' no longer surfaces in a species-filtered search.

File: apps/backend/src/services/atcvet.service.ts [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+  species,
+  limit = 20,
+}: SuggestMedicationsParams): Prisma.Sql => {
+  const term = q?.trim() ?? "";
+  const contains = `%${escapeLike(term)}%`;
+  const prefix = `${escapeLike(term)}%`;
+
+  const search = term
+    ? Prisma.sql`AND (e."display" ILIKE ${contains} ESCAPE '\\' OR e."code" ILIKE ${prefix} ESCAPE '\\')`
+    : Prisma.empty;
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/src/services/atcvet.service.ts:78]: Correct, and this one was my own trap - I added the partial GIN index precisely so medication search would not scan, then queried e.display directly, which cannot use it. The predicate now searches code_entry_search_text(display, synonyms), the same expression the index is built on. The code column is matched separately on purpose: it is not part of the indexed text, and a prefix match on a short code column is cheap.

File: apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx [hunk 10]
```diff
@@ -391,6 +395,47 @@ const PrescriptionRow = ({
...
+    requestSeq.current = requestId;
+    const timer = setTimeout(
+      () => {
+        if (readOnly || trimmed.length < ATCVET_MIN_QUERY) {
+          setItems([]);
+          return;
+        }
+        suggestMedications({ q: trimmed, limit: ATCVET_LIMIT })
+          .then((results) => {
+            if (requestSeq.current === requestId) setItems(results);
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx]: Fixed, and the first attempt was wrong in an instructive way: clearing synchronously in the effect tripped react-hooks/set-state-in-effect (caught by the pre-commit hook). Rewritten to derive instead of store - the page is kept with the query that produced it and rendered only while that query still matches. Both staleness rules are now structural rather than bookkeeping: a page for an older query never renders, whether it arrives late or is left over from the previous keystroke.

File: apps/backend/scripts/convert-atcvet-xlsx.mjs [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+fs.writeFileSync(
+  destination,
+  JSON.stringify(
+    {
+      source: "WHO Collaborating Centre for Drug Statistics Methodology",
+      dataset: "ATCvet index",
+      // Stamped from the filename rather than invented, so provenance on every
+      // imported row names the release it actually came from.
+      release: /(\d{4})/.exec(source)?.[1] ?? "unknown",
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/scripts/convert-atcvet-xlsx.mjs]: Fixed - the regex now runs on the filename only, so a path like ~/archive/2019/2026 ATCvet index.xlsx is stamped 2026 rather than 2019. Verified by re-running the converter against the real workbook: release 2026, 8315 entries.

File: apps/backend/src/models/code-entry.ts [hunk 12]
```diff
@@ -1,4 +1,5 @@
-export type CodeSystem = "YOSEMITECODE" | "IDEXX" | "SNOMED" | "VENOM";
+export type CodeSystem =
+  "YOSEMITECODE" | "IDEXX" | "SNOMED" | "VENOM" | "ATCVET";
 export type CodeType =
   | "SPECIES"
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/src/models/code-entry.ts:4]: Fixed - MEDICATION and MEDICATION_CATEGORY added to the backend CodeType union, which had drifted from the Prisma enum. I had updated the CodeSystem union in the same file for ATCVET and missed its neighbour.

File: apps/backend/src/scripts/import-atcvet-index.ts [hunk 13]
```diff
@@ -0,0 +1,248 @@
...
+      skipped.push({ code, reason: "duplicate code in extract" });
+      continue;
+    }
+    seen.add(code);
+    const level = LEVEL_BY_LENGTH[code.length];
+    entries.push({
+      code,
+      display,
+      // Only the substance level is prescribable; the four above it are headings.
+      type: level === 5 ? "MEDICATION" : "MEDICATION_CATEGORY",
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/src/scripts/import-atcvet-index.ts:134]: Declining this one, with the numbers. Of 6,417 substance rows, 80 carry a non-specific label (42 literally "combinations", 6 "various", the rest "combinations of sulfonamides" and similar). They are genuine ATCvet substance-level codes - a combination product legitimately IS QA01AA30 - so excluding them from the import would make those products uncodable, which is the failure this PR exists to remove.

The readability concern is real but already handled by the picker: every suggestion renders with its class path, so such a row reads "combinations · Caries prophylactic agents · STOMATOLOGICAL PREPARATIONS" rather than a bare word. Ranking also keeps them out of the way - exact code match first, then display-prefix, then alphabetical - so they only appear when someone actually types toward them.

Happy to revisit if these prove noisy in the dev QA pass; the cheap follow-up would be down-ranking non-specific labels rather than dropping the codes.

**Review by aupyay — DISMISSED** [review]

File: apps/backend/src/scripts/import-atcvet-index.ts [hunk 14]
```diff
@@ -0,0 +1,275 @@
...
+
+  // A newer yearly release can withdraw a code. Rows absent from this extract are
+  // deactivated rather than deleted: a prescription or formulary row may already
+  // reference one, and a dangling code with no entry is worse than a retired one
+  // that can still be displayed.
+  const retired = await prisma.codeEntry.updateMany({
+    where: {
+      system: "ATCVET",
+      active: true,
+      code: { notIn: plan.entries.map((entry) => entry.code) },
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/src/scripts/import-atcvet-index.ts]: Right, and this was created by the retirement fix from your previous round - thank you for catching that it made things worse. Retirement now runs only against something that looks like a complete release: at least 5,000 codes AND no more than 10% smaller than what is currently active (the index has held ~8,300 for years and a yearly release moves a handful). Otherwise it logs SKIP retirement with both counts and imports without deactivating anything. Test asserts a one-entry extract retires nothing and says so.

File: apps/frontend/src/app/features/appointments/lib/inventoryPrescription.ts [hunk 15]
```diff
@@ -211,6 +214,7 @@ export const inventoryToPrescriptionItem = (item: InventoryItem): Omit<Prescript
     medicineName: basicInfo.name,
     brand: cleanString(basicInfo.brand ?? classification.brand),
     genericName: cleanString(classification.genericName),
+    atcCode: cleanString(item.atcCode),
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/frontend/src/app/features/appointments/lib/inventoryPrescription.ts:217]: Confirmed and fixed - and this is the second time this feature dead-ended one layer further up, which is worth naming: last round the field was missing from inventoryToPrescriptionItem, this round from mapApiItemToInventoryItem, so items coming from fetchInventoryItems still arrived without a code. atcCode is now copied there (falling back to the attributes bag like genericName does), so the path from backfill to prescription line is complete.

File: apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx [hunk 16]
```diff
@@ -391,6 +395,52 @@ const PrescriptionRow = ({
...
+  const skip = readOnly || trimmed.length < ATCVET_MIN_QUERY;
+
+  useEffect(() => {
+    if (skip) return;
+    const timer = setTimeout(() => {
+      suggestMedications({ q: trimmed, limit: ATCVET_LIMIT })
+        .then((items) => setPage({ query: trimmed, items }))
+        .catch((error) => {
+          console.error('Unable to suggest medications:', error);
+          setPage({ query: trimmed, items: [] });
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx]: Correct, and a regression I introduced with the keyed-page rewrite. Fixed by cancelling on effect cleanup: a superseded request cannot publish at all, so it can neither overwrite the newer page nor cause the render guard to hide both. Keying by query alone was genuinely not enough, exactly as you describe. New test drives A-pending, B-resolves, A-completes and asserts B's results survive.

File: apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx [hunk 17]
```diff
@@ -391,6 +395,52 @@ const PrescriptionRow = ({
...
+    items: [],
+  });
+
+  const trimmed = query.trim();
+  const skip = readOnly || trimmed.length < ATCVET_MIN_QUERY;
+
+  useEffect(() => {
+    if (skip) return;
+    const timer = setTimeout(() => {
+      suggestMedications({ q: trimmed, limit: ATCVET_LIMIT })
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx]: Fixed by threading it properly: companion.species now flows workspace -> TreatmentStep -> PrescriptionEditor -> suggestMedications, so the species metadata the importer attaches is finally used at the point of prescribing. Two tests: the filter is sent when the species is known, and omitted when it is not (an unknown species must not narrow the list). Worth noting the endpoint keeps species-agnostic substances in a filtered search, so this excludes other-species vaccines without hiding general drugs.

File: apps/backend/src/scripts/backfill-atc-codes.ts [hunk 18]
```diff
@@ -0,0 +1,209 @@
...
+    .trim();
+
+/**
+ * Strips a trailing dose/pack description so "cefalexin 250 mg tablet" can match
+ * the substance "cefalexin". Only a trailing run is removed: a leading number is
+ * part of a name (as in "5 fluorouracil") and must survive.
+ */
+export const stripPresentation = (value: string): string =>
+  normalise(value)
+    .replace(/\s+\d+(\s*\.\d+)?\s*(mg|g|ml|mcg|iu|%|kg|mg\/ml|mg\/kg)\b.*$/, "")
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/src/scripts/backfill-atc-codes.ts]: Good catch - fixed. normalise has already turned "1%" into "1" by the time stripPresentation runs, so requiring a unit left "hydrocortisone 1 cream" and matched nothing. The unit is now optional in that expression. Tests cover Hydrocortisone 1% cream and Enrofloxacin 2.5 % Injection, and the leading-number case (5-fluorouracil) still passes, which is what stops the looser pattern eating a name.

File: apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx [hunk 19]
```diff
@@ -485,6 +537,41 @@ const PrescriptionEditor = ({
...
+                      ) : undefined
+                    }
+                    onSelect={() => {
+                      onAddItem({
+                        medicineName: substance.label,
+                        atcCode: substance.atcCode,
+                        // A substance picked from the classification is not
+                        // stock, so it cannot be dispensed in-house: it is
+                        // written for the owner to have filled elsewhere.
+                        fulfillment: 'PRESCRIPTION_ONLY',
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/frontend/src/app/features/appointments/pages/AppointmentWorkspace/components/PrescriptionEditor.tsx:603]: Fixed. A line with an atcCode but no inventoryItemId and no SKU is classification-only, and its fulfillment control is now disabled, so it cannot be switched to in-house and finalized into a dispense request against stock that does not exist. The same substance backed by real stock keeps the choice - covered by a test asserting disabled in one case and enabled in the other.

File: packages/database/prisma/schema.prisma [hunk 20]
```diff
@@ -5832,6 +5845,10 @@ model DrugFormulary {
   organisationId String
   drugName       String
   genericName    String?
+  /// ATCvet substance code (CodeEntry.system = ATCVET, type = MEDICATION).
+  /// Nullable and never guessed: an unmatched or ambiguous drug stays uncoded
+  /// rather than being filed under the wrong therapeutic class.
+  atcCode        String?
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply packages/database/prisma/schema.prisma:5851]: Fixed - atcCode added to formularySelect, so the backfilled code is actually readable through the formulary GET/list rather than sitting write-only in the column.

File: apps/backend/package.json [hunk 21]
```diff
@@ -20,6 +20,8 @@
     "seed:template-library": "tsx src/scripts/seed-template-library.ts",
     "check:wallet": "tsx src/scripts/check-wallet-config.ts",
     "import:venom-designations": "tsx src/scripts/import-venom-designations.ts",
+    "backfill:atc-codes": "tsx src/scripts/backfill-atc-codes.ts",
+    "import:atcvet-index": "tsx src/scripts/import-atcvet-index.ts",
```
[the remark this answers is not in the document]
Comment (ankit-yc) [reply apps/backend/package.json:24]: Correct that a deploy alone leaves the vocabulary empty - addressed in f6ed365, deliberately as a documented rollout step rather than by wiring the import into api-deploy.sh.

Reasoning: the index changes once a year while the API deploys many times a week, so running an 8,315-row import on every deploy would add minutes to each one to rewrite rows that have not changed, and would let a bad data file ride out with an ordinary code deploy. It also matches how the VeNom hierarchy and designation imports were rolled out.

docs/guide/atcvet-rollout.md now gives the exact commands (dry run first, then --apply), the expected output for the 2026 release, the yearly re-import path, the retirement safety rule, and a curl to verify substances are searchable afterwards. The importer header points at it so anyone reading the script finds the runbook.

If you would still rather it ran automatically, the cheap version is a guarded step in api-deploy.sh that skips when the active ATCVET count already matches the extract - happy to add that as a follow-up if the team prefers it.
